### PR TITLE
fix(storage): set flush and get_state to false on the last write in gRPC

### PR DESCRIPTION
**Pull request by BrennaEpp**
Only receive if get_state was true (ie. `remainingDataFitsInSingleReq && !lastWriteOfEntireObject`) (or after CloseSend) to avoid blocking.

### CODE REVIEW

File: storage/grpc_client.go [hunk 1]
```diff
@@ -1724,31 +1724,31 @@ func (w *gRPCWriter) uploadBuffer(recvd int, start int64, doneReading bool) (*st
 		}
 
 		// Done sending data (remainingDataFitsInSingleReq should == true if we
```
Comment (tritone) [inline storage/grpc_client.go:1726]: Maybe clarify that we are done sending data in the provided buffer, but there are two separate cases depending on whether there is more data to send in the object overall. It's a little confusing as currently written.

File: storage/grpc_client.go [hunk 2]
```diff
@@ -1724,31 +1724,31 @@ func (w *gRPCWriter) uploadBuffer(recvd int, start int64, doneReading bool) (*st
...
-		}
-		if err != nil {
-			return nil, 0, err
-		}
-
-		// Confirm the persisted data if we have not finished uploading the object.
-		if !lastWriteOfEntireObject {
 			if resp.GetPersistedSize() != writeOffset {
 				// Retry if not all bytes were persisted.
 				writeOffset = resp.GetPersistedSize()
```
Comment (tritone) [inline storage/grpc_client.go:1754]: Is it actually the case that we no longer need to revc() from the stream if we are ready to finalize the object on this msg?

Comment (BrennaEpp) [reply storage/grpc_client.go:1754]: We do still receive (see L1775), but only until after `CloseSend`.

Comment (tritone) [reply storage/grpc_client.go:1755]: Ah I see, makes sense.

File: storage/grpc_client.go (hunk 1 above)
Comment (BrennaEpp) [reply storage/grpc_client.go]: Modified the wording, does it make sense now?

**Review by tritone — APPROVED** [review]

**Comment by tritone** [comment]
Also, commit type could be `fix` for this PR.

**Pull request merged**